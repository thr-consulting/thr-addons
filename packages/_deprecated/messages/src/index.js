import {
	addMessage,
	addErrorMessage,
	addSuccessMessage,
	addWarnMessage,
	clearMessage,
	clearMessages,
} from './messagesActions';
import messagesReducer from './messagesReducer';
import Messages from './Messages';

/**
 * Clears all messages if messages exist. Call from route.onChange().
 * @param store - Redux store
 */
function clearAllMessages(store) {
	const msgs = store.getState().get('messages');
	if (msgs && msgs.size > 0) store.dispatch(clearMessages());
}

export {
	addMessage,
	addErrorMessage,
	addSuccessMessage,
	addWarnMessage,
	clearMessage,
	clearMessages,
	clearAllMessages,
	messagesReducer,
	Messages,
};
