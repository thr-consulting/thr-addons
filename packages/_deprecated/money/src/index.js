import getCountryCode from './currencyCountry';
import GraphQLMoney from './graphql/GraphQLMoney';
import {makeMoney, formatMoney, roundTo, transformObjectsToMoney} from './util';

export {
	getCountryCode,
	GraphQLMoney,
	makeMoney,
	formatMoney,
	roundTo,
	transformObjectsToMoney,
};

/**
 * A JS-Money object.
 * @typedef Money
 */
