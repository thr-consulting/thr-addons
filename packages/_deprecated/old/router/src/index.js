import Reroute from './Reroute';
import SwitchWithError from './SwitchWithError';
import RouteDirector from './RouteDirector';
import lazy from './lazy';

export {
	Reroute,
	SwitchWithError,
	RouteDirector,
	lazy,
};
