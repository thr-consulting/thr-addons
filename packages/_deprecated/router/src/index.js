import ContainerLayout from './ContainerLayout';
import FullWidthLayout from './FullWidthLayout';
import SimpleLayout from './SimpleLayout';
import Reroute from './Reroute';
import SwitchWithError from './SwitchWithError';

export {
	ContainerLayout,
	FullWidthLayout,
	SimpleLayout,
	Reroute,
	SwitchWithError,
};
