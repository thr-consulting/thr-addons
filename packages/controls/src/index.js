/**
 * A collection of form components designed with SemanticUI. They all adhere to React Formal's policy of having
 * a value prop and an onChange() prop.
 * */

import {Field, Button as SubmitButton} from 'react-formal';
import SForm from './SForm';
import ResponsiveMenu from './ResponsiveMenu';
import FieldMap from './FieldMap';
import RadioGroup from './RadioGroup';
import MaskedInput from './MaskedInput';
import SinEntry from './SinEntry';

export {
	SForm,
	ResponsiveMenu,
	FieldMap,
	RadioGroup,
	Field,
	SubmitButton,
	MaskedInput,
	SinEntry,
};
