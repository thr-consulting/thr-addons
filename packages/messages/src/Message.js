// @flow

/* eslint-disable jsx-a11y/no-static-element-interactions */
import React from 'react';
import {bindActionCreators} from 'redux';
import {connect} from 'react-redux';
import {Message as SMessage, Icon, List} from 'semantic-ui-react';
import {clearMessage} from './messagesActions';

type Props = {
	clearMessage: (id: string) => {},
	message: {
		id: string,
		level: string,
		message: string | string[],
		title: string,
	},
};

export function MessageComponent(props: Props) {
	const warning = props.message.level === 'warning';
	const error = props.message.level === 'error';
	const info = props.message.level === 'info';
	const success = props.message.level === 'success';
	if (Array.isArray(props.message.message)) {
		return (
			<SMessage warning={warning} error={error} info={info} success={success}>
				<Icon name="close" onClick={() => props.clearMessage(props.message.id)}/>
				<SMessage.Header>{props.message.title}</SMessage.Header>
				<List bulleted>
					{props.message.message.map(message => <List.Item key={message}>{message}</List.Item>)}
				</List>
			</SMessage>
		);
	}
	return (
		<SMessage warning={warning} error={error} info={info} success={success}>
			<Icon name="close" onClick={() => props.clearMessage(props.message.id)}/>
			<SMessage.Header>{props.message.title}</SMessage.Header>
			<p>{props.message.message}</p>
		</SMessage>
	);
}

const mapDispatchToProps = dispatch => bindActionCreators({
	clearMessage,
}, dispatch);

export default connect(null, mapDispatchToProps)(MessageComponent);
