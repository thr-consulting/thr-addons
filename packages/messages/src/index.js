import {
	addMessage,
	addErrorMessage,
	addSuccessMessage,
	addWarnMessage,
	clearMessage,
	clearMessages,
} from './messagesActions';
import messagesReducer from './messagesReducer';
import Messages from './Messages';

/**
 * Clears all messages if messages exist. Call from route.onChange().
 * @param store - Redux store
 */
function clearAllMessages(store) {
	if (store.getState().get('messages').size > 0) store.dispatch(clearMessages());
}

export {
	addMessage,
	addErrorMessage,
	addSuccessMessage,
	addWarnMessage,
	clearMessage,
	clearMessages,
	clearAllMessages,
	messagesReducer,
	Messages,
};
