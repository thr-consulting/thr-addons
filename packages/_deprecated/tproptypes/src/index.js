// @flow

import {localDate} from './date';
import {money} from './money';
import {reactElements} from './react';
import {history, location} from './router';
import {sFieldMessages} from './sform';

export default {
	location,
	history,
	reactElements,
	sFieldMessages,
	money,
	localDate,
};
