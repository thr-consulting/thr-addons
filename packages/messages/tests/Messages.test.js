import React from 'react';
import {shallow} from 'enzyme';
import {Map} from 'immutable';
import {MessagesComponent} from '../src/Messages';

const messages = new Map({
	default: {
		id: 'myid',
		level: 'success',
		message: 'This is a message',
		title: 'This is a title',
	},
});

describe('Messages', () => {
	it('should render without throwing an error', () => {
		expect(shallow(
			<MessagesComponent messages={messages}/>
		)).toMatchSnapshot();
	});

	it('should render empty if no messages are given', () => {
		expect(shallow(<MessagesComponent messages={new Map()}/>)).toMatchSnapshot();
	});
});
