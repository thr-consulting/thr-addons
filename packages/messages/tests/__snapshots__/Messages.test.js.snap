// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Messages should render null if no messages are given 1`] = `null`;

exports[`Messages should render without throwing an error 1`] = `
<GridRow>
  <GridColumn>
    <Connect(MessageComponent)
      message={
        Object {
          "id": "myid",
          "level": "success",
          "message": "This is a message",
          "title": "This is a title",
        }
      }
    />
  </GridColumn>
</GridRow>
`;
